mlx/traceable_attribute.py

```diff
@@ -0,0 +1,54 @@
+'''
+Storage class for traceable item attribute
+'''
+
+import re
+from mlx.traceable_base_class import TraceableBaseClass
+
+
+class TraceableAttribute(TraceableBaseClass):
+    '''
+    Storage for an attribute to a traceable documentation item
+    '''
+
+    def __init__(self, attrid, value):
+        '''
+        Initialize a new attribute
+
+        Args:
+            attrid (str): Attribute identification
+            value (str): Regex to which the attribute values should match
+        '''
+        super(TraceableAttribute, self).__init__(attrid)
+        self.value = value
+
+    @staticmethod
+    def to_id(id):
+        '''
+        Convert a given identification to a storable id
+
+        Args:
+            id (str): input identification
+        Returns:
+            str - Converted storable identification
+        '''
+        return id.lower()
```

Comment:
I would really put a warning that we do this - otherwise we might cause users a headache when `Something` will be same as `something`

Reply:
The readme has a not about it